contracts/contracts/strategies/NativeStaking/NativeStakingSSVStrategy.sol

```diff
@@ -0,0 +1,264 @@
+    /// @param _asset WETH to withdraw
+    /// @param _amount Amount of WETH to withdraw
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        _withdraw(_recipient, _asset, _amount);
+    }
+
+    function _withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) internal {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient != address(0), "Must specify recipient");
+
+        emit Withdrawal(_asset, address(0), _amount);
+        IERC20(_asset).safeTransfer(_recipient, _amount);
+    }
+
+    /// @notice transfer all WETH deposits back to the vault.
+    /// This does not withdraw from the validators. That has to be done separately with the
+    /// `exitSsvValidator` and `removeSsvValidator` operations.
+    /// This does not withdraw any execution rewards from the FeeAccumulator or
+    /// consensus rewards in this strategy.
+    /// Any ETH in this strategy that was swept from a full validator withdrawal will not be withdrawn.
+    /// ETH from full validator withdrawals is sent to the Vault using `doAccounting`.
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    function withdrawAll() external override onlyVaultOrGovernor nonReentrant {
+        uint256 wethBalance = IERC20(WETH_TOKEN_ADDRESS).balanceOf(
+            address(this)
+        );
+        if (wethBalance > 0) {
+            _withdraw(vaultAddress, WETH_TOKEN_ADDRESS, wethBalance);
+        }
+    }
+
+    function _abstractSetPToken(address _asset, address) internal override {}
+
+    /// @notice Returns the total value of (W)ETH that is staked to the validators
+    /// and WETH deposits that are still to be staked.
+    /// This does not include ETH from consensus rewards sitting in this strategy
+    /// or ETH from MEV rewards in the FeeAccumulator. These rewards are harvested
+    /// and sent to the Dripper so will eventually be sent to the Vault as WETH.
+    /// @param _asset      Address of weth asset
+    /// @return balance    Total value of (W)ETH
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256 balance)
+    {
+        require(_asset == WETH_TOKEN_ADDRESS, "Unsupported asset");
+
+        balance =
+            // add the ETH that has been staked in validators
+            activeDepositedValidators *
+            32 ether +
+            // add the WETH in the strategy from deposits that are still to be staked
+            IERC20(WETH_TOKEN_ADDRESS).balanceOf(address(this));
+    }
+
+    function pause() external onlyStrategist {
+        _pause();
+    }
+
+    /// @notice Returns bool indicating whether asset is supported by strategy.
+    /// @param _asset The address of the asset token.
+    function supportsAsset(address _asset) public view override returns (bool) {
```

Comment:
This function can be set to external.

Reply:
this one overrides the function from `InitializableAbstractStrategy.sol` and can not have a different visibility